### ISSUE TITLE
`keyring.delete_password(service, username)` deletes all passwords for that `service`

### ISSUE THREAD

**Issue by jaraco** [jaraco/keyring#123]
In Ubuntu 12.04 using keyring3.1

```
#!python

>>> import keyring

>>> keyring.set_password('jacks', 'cold', 'sweat')
>>> keyring.get_password('jacks', 'cold')
u'sweat'
>>> keyring.set_password('jacks', 'smirking', 'revenge')
>>> keyring.get_password('jacks', 'smirking')
u'revenge'
>>> keyring.delete_password('jacks', 'cold')
>>> keyring.get_password('jacks', 'smirking')
>>>

```

It seems that all passwords for the given service are deleted when a `delete_password` call is made. 

This does not happen in OS X (10.8.5)

---

I guess the discrepancy would lie between these two methods. 

[keyring.backends.OS_X.Keyring.delete_password](https://bitbucket.org/kang/python-keyring-lib/src/3.1/keyring/backends/OS_X.py?at=default#cl-103)

[keyring.backends.file.BaseKeyring](https://bitbucket.org/kang/python-keyring-lib/src/3.1/keyring/backends/file.py?at=default#cl-123)

---
- Bitbucket: https://bitbucket.org/kang/python-keyring-lib/issue/123
- Originally reported by: Sam Stavinoha
- Originally created at: 2013-10-24T18:42:03.255

**Comment by jaraco** [comment]
Will have a fix for that soon.

---

Original comment by: Dmitry Shachnev

**Comment by jaraco** [comment]
Thanks Dmitry. Indeed, delete_password should only ever delete a single password. Perhaps we should build a test for that into the base test helper.

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Fixed in pull request #38.

---

Original comment by: Dmitry Shachnev

**Comment by jaraco** [comment]
Merged in mitya57/python-keyring-lib (pull request #38)

Fix items deleting in file keyring (Fixes #123), add a test for that

→ <<cset fc974ee32eaa>>

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Fix items deleting in file keyring (Fixes #123), add a test for that

→ <<cset cc6a363b447c>>

---

Original comment by: Jason R. Coombs
